moped-editor/src/views/projects/projectView/ProjectTeam/TeamAutocompleteComponent.js

```diff
@@ -35,46 +40,36 @@ const TeamAutocompleteComponent = ({
   }, [hasFocus]);
 
   const handleChange = (event, newValue) => {
-    const personnelValue = nameLookup[newValue];
     apiRef.current.setEditCellValue({
       id,
       field,
-      value: personnelValue ?? null,
+      value: newValue,
     });
     // Also update the corresponding workgroup field with the selected user's workgroup id
     apiRef.current.setEditCellValue({
       id,
       field: "moped_workgroup",
-      value: { workgroup_id: userWorkgroupLookup[newValue] },
+      value: { workgroup_id: newValue?.workgroup_id },
     });
   };
 
-  const options = Object.keys(nameLookup);
-
   const isOptionEqualToValue = (option, value) => {
-    // if the value is a number, use the idFromValue nameLookup to find if option is equal to Value
-    // If the value is an object, use the user_id to find if option is equal to Value
-    let idFromValue;
-    if (typeof value === "string") {
-      idFromValue = Object.keys(nameLookup).find(
-        (key) => nameLookup[key] === value
-      );
-    } else if (typeof value === "object") {
-      idFromValue = value.user_id;
-    }
-
-    if (Number(option) === Number(idFromValue)) {
-      return true;
-    } else {
-      return false;
-    }
+    return value?.user_id === option?.user_id;
   };
 
   const getOptionLabel = (option) => {
-    if (typeof option === "string" && !nameLookup[option]) {
-      return option;
-    }
-    return nameLookup[option] || "";
+    return option.user_id ? `${option.first_name} ${option.last_name}` : "";
+  };
+
+  const renderOption = (props, option) => {
+    return (
+      <ListItem {...props} key={option.user_id}>
+        <ListItemText
+          primary={`${option.first_name} ${option.last_name}`}
+          secondary={option.email}
+        />
+      </ListItem>
+    );
```

Comment:
This looks like it can/should exist as a freestanding component, like `<TeamMemberOption/>` or whatever.